shardingsphere-kernel/shardingsphere-single-table/shardingsphere-single-table-core/src/test/java/org/apache/shardingsphere/singletable/route/SingleTableSQLRouterTest.java

```diff
@@ -104,6 +104,24 @@ public void assertCreateRouteContextWithMultiDataSource() {
         assertFalse(actual.isFederated());
     }
     
+    @Test
+    public void assertDecorateRouteContext() {
+        SingleTableRule singleTableRule = new SingleTableRule(new SingleTableRuleConfiguration(), DefaultDatabase.LOGIC_NAME, mock(DatabaseType.class),
+                createSingleDataSourceMap(), Collections.emptyList(), new ConfigurationProperties(new Properties()));
+        singleTableRule.getSingleTableDataNodes().put("t_order", Collections.singletonList(mockDataNode("ds_0")));
+        ShardingSphereMetaData metaData = mockSingleDataSourceMetaData();
+        RouteContext actual = new RouteContext();
```

Comment:
Hi @hoorf, please check the `decorateRouteContext` pre-logic, the `decorateRouteContext` method will be executed only when the RouteContext contains `rountUnits`. Also, since there are multiple implementations of SQLRouter, consider testing in multiple combined scenarios.